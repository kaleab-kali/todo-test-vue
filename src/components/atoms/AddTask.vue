<template>
  <div :class="$style.add">
    <input
      type="text"
      placeholder="Add new task"
      @keyup.enter="clearInput"
      v-model="value"
    />
  </div>
</template>

<script>
export default {
  name: "AddTask",
  props: {
    value: "",
  },
  methods: {
    clearInput() {
      this.$emit("keyup", this.value);
      this.value = "";
    },
  },
};
</script>

<style lang="scss" module>
.add {
  font-family: "Inter";
  padding: 0.688rem 0;
  border-radius: 0.625rem;
  display: flex;
  color: $lightOrange;
  background-color: $peach;
  border: 2px dashed $lightOrange;
  max-width: 30rem;
  margin-top: 1rem;
  @media (max-width: 980px) {
    max-width: 30rem;
    margin: 1rem;
  }
  @media (max-width: 450px) {
    margin-bottom: 0;
  }
  input {
    font-weight: 400;
    font-size: 1.25rem;
    color: $lightOrange;
    background-color: $peach;
    border: none;
    outline: none;
    display: flex;
    text-align: center;
    @media (max-width: 450px) {
      max-width: 10rem;
      font-size: 1rem;
    }
  }
}
.add:hover {
  border: 2px dashed $orange;
}
.add:active {
  border: 2px solid $orange;
}
</style>
