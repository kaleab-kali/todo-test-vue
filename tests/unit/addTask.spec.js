import { mount } from "@vue/test-utils";
import AddTask from "@/components/atoms/AddTask.vue";

describe("Atom AddTask Component", () => {
  it("correct layout", () => {
    const wrapper = mount(AddTask);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Task is entered", async () => {
    const wrapper = mount(AddTask);
    await wrapper.find("input").trigger("keyup.enter");
    await wrapper.find("input").trigger("change");
    expect(wrapper.emitted().keyup).toBeTruthy();
    expect(wrapper.emitted().change).toBeTruthy();
  });
});
