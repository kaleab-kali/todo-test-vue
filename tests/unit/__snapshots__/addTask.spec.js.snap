// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask.vue has input 1`] = `<div class="add"><input type="text" placeholder="Add new task"></div>`;

exports[`Atom AddTask Component correct layout 1`] = `<div class="add"><input type="text" placeholder="Add new task"></div>`;

exports[`Atom AddTask Component has input 1`] = `<div class="add"><input type="text" placeholder="Add new task"></div>`;
